source/CIGameUpdateManager.cpp

```diff
@@ -49,6 +49,18 @@ bool GameUpdateManager::init() {
     return true;
 }
 
+/**
+ * Resets the manager on game reset.
+ */
+void GameUpdateManager::reset() {
```

Comment:
This class definitely does not need a reset method. Just use dispose and alloc a new one when a new game starts